### DISCUSSION TITLE
2.4.10 fleet server and these packages too old, how to update ?

### DISCUSSION THREAD

**Discussion by StefanSa** [Security-Onion-Solutions/securityonion#11454] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

upgrading

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

12

### RAM

128GB

### Storage for /

100GB

### Storage for /nsm

2TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

No, there are no additional clues

### Detail

Hi there
have here the last version of so v2.4.10
But here is the fleet server and its packages too old.
Without the latest fleet packages and agents i can't get the firewall logs parsed from Sophos utm/xg.

Question,
how do i update the fleedserver and its packages ?

Thanks for any help.
Stefan

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by defensivedepth** [comment]
Hello there - 

` But here is the fleet server and its packages too old.`

Please provide more detail. What version are you looking for?

**Comment by StefanSa** [comment]
hi @defensivedepth 
Here is the appropriate discussion on why it generally does not work in this older version.
https://github.com/elastic/integrations/issues/8022

Currently there is `version 3.4.0` for this Sophos integration, `so` has a significantly older version with `2.11.0`
https://docs.elastic.co/integrations/sophos#changelog

There should be a possibility `to update the fleet server` and therefore also `automatically the corresponding integration`.

**Reply by defensivedepth** [reply]
Version `2.11.0` was initially released as part of Elastic 8.8.2, June 29th, 2023. Security Onion is currently using 8.8.2 - When we upgrade to the next major release of Elastic, these packages will be upgraded as well. We are not on the bleeding edge of Elastic releases because of stability concerns.

**Comment by StefanSa** [comment]
@defensivedepth Josh
I can agree with that -> `We are not on the bleeding edge of Elastic releases because of stability concerns.`
But what do you do when the source changes (in this case syslog firewall), other fields other mapping. Then you can't analyze / receive data correctly with this old fleet integration. Therefore there must be the possibility to keep the integrations up to date or at least to leave the decision to the user.
But the fact is that at least this Sophos integration in version 2.11.0, possibly others have problems with the data pipeline. It seems that on the way from the elastic agent via logstash the pipeline is not transfered correctly.
The pipeline itself works without any problem.
You can read more about it here:
https://github.com/elastic/integrations/issues/8022

There are also users who have exactly the same problem.
Currently i help myself with a standalone filebeat instance, because at least this fleet integration doesn't work correctly.
https://github.com/elastic/integrations/issues/8022#issuecomment-1752922448

I haven't found time yet to do a trace via wireshark to understand why there are problems with the pipeline or similar.

PS: Quick info.
i am one of the initial contributors of the sophos firewall module from elastic beats.

Best Regards
Stefan

**Reply by defensivedepth** [reply, marked as answer]
In order to automate & integrate the platform like we have, it is required to keep the versions in sync.  

I have created an issue to track upgrading the Sophos integration - which should happen automatically next time we upgrade the Elastic stack - https://github.com/Security-Onion-Solutions/securityonion/issues/11548
